SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestHandoverReplication/SpatialTestHandoverReplication.cpp

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestHandoverReplication.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Kismet/GameplayStatics.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "DynamicReplicationHandoverCube.h"
+#include "SpatialFunctionalTestFlowController.h"
+
+/**
+ * This tests that an Actor's bReplicate flag is properly handed over when dynamically set.
+ * Tests UNR-4441
+ * This test contains 4 Server and 2 Client workers.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *	- Server 1 spawns a HandoverCube with bReplicates set to false inside its authority area. (ADynamicReplicationHandoverCube)
+ *	- The bReplicates flag is set to true after the end of the Actor's initialization.
+ *  - All servers set a reference to the HandoverCube and reset their local copy of the LocationIndex and AuthorityCheckIndex.
+ * - Test:
+ *	- At this stage, Server 1 should have authority over the HandoverCube.
+ *  - The HandoverCube is moved in the authority area of Server 2.
+ *  - At this stage, Server 2 should have authority over the HandoverCube.
+ *  - Server 2 acquires a lock on the HandoverCube and moves it into the authority area of Server 4.
+ *	- Since Server 2 has the lock on the HandoverCube it should still be authoritative over it.
+ *  - Server 2 releases the lock on the HandoverCube.
+ *  - At this point, Server 4 should become authoritative over the HandoverCube.
+ *  - The HandoverCube is moved in the authority area of Server 3.
+ *  - At this point, Server 3 should be authoritative over the Handover	Cube.
+ * - Clean-up:
+ *	- The HandoverCube is destroyed.
+ */
+
+ASpatialTestHandoverReplication::ASpatialTestHandoverReplication()
+	: Super()
+{
+	Author = "Antoine Cordelle";
+	Description = TEXT("Test dynamically set replication for an actor");
+
+	Server1Position = FVector(-500.0f, -500.0f, 50.0f);
+	Server2Position = FVector(500.0f, -500.0f, 50.0f);
+	Server3Position = FVector(-500.0f, 500.0f, 50.0f);
+	Server4Position = FVector(500.0f, 500.0f, 50.0f);
+}
+
+void ASpatialTestHandoverReplication::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("Server spawns one DynamicReplicationHandoverCube"), FWorkerDefinition::Server(1), nullptr, [this]() {
```

Comment:
```suggestion
	AddStep(TEXT("Server 1 spawns a HandoverCube (called ADynamicReplicationHandoverCube) with bReplicates set to false inside its authority area."), FWorkerDefinition::Server(1), nullptr, [this]() {
```

I think it's good practice if the TEXT descriptor of each step matches the steps outlined in the `The flow is as follows:` section.

If you agree:
1. Accept this.
1. Do the same for each step
1. Take a look at https://brevi.link/how-to-test-unrealgdk and, if this formal preference isn't made explicit in that guide, please make it so.